python/hail/expr/expression.py

```diff
@@ -529,6 +530,157 @@ def __ne__(self, other):
+            df = hail.Table.range(1) # wat
+            df = df.select(**{'<expr>': self})
+            return df
+        elif len(axes) == 0:
+            source = source.select_globals(**{'<expr>': self})
+            df = hail.Table.range(1)
+            df = df.select(**{'<expr>': source.view_join_globals()['<expr>']})
+            return df
+        elif len(axes) == 1:
+            if isinstance(source, hail.Table):
+                df = source
+                df = df.select(*df.key, **{'<expr>': self}).select_globals()
+                return df
+            else:
+                assert isinstance(source, hail.MatrixTable)
+                if self._indices == source._row_indices:
+                    source = source.select_rows(*source.row_key, **{'<expr>': self})
+                    return source.rows_table()
+                else:
+                    assert self._indices == source._col_indices
+                    source = source.select_cols(*source.col_key, **{'<expr>': self})
+                    return source.cols_table()
+        else:
+            assert len(axes) == 2
+            assert isinstance(source, hail.MatrixTable)
+            source = source.select_entries(**{'<expr>': self})
+            source = source.select_rows(*source.row_key)
+            source = source.select_cols(*source.col_key)
+            return source.entries_table()
+
+    @handle_py4j
+    @typecheck_method(n=int, width=int, truncate=nullable(int), types=bool)
+    def show(self, n=10, width=90, truncate=None, types=True):
+        """Print the first few rows of the table to the console.
+
+        Examples
+        --------
+        .. doctest::
+
+            >>> table1.SEX.show()
+            +-------+--------+
+            |    ID | SEX    |
+            +-------+--------+
+            | Int32 | String |
+            +-------+--------+
+            |     1 | M      |
+            |     2 | M      |
+            |     3 | F      |
+            |     4 | F      |
+            +-------+--------+
+
+            >>> hl.capture(123).show()
+            +--------+
+            | <expr> |
+            +--------+
+            |  Int32 |
+            +--------+
+            |    123 |
+            +--------+
+
+        Warning
+        -------
+        Extremely experimental.
+
+        Parameters
+        ----------
+        n : :obj:`int`
+            Maximum number of rows to show.
+        width : :obj:`int`
+            Horizontal width at which to break columns.
+        truncate : :obj:`int`, optional
+            Truncate each field to the given number of characters. If
+            ``None``, truncate fields to the given `width`.
+        types : :obj:`bool`
+            Print an extra header line with the type of each field.
+        """
+        name = None
+        source = self._indices.source
+        if source is not None:
+            fields_rev = {v: k for k, v in source._fields.items()}
+            name = fields_rev.get(self, None)
```

Comment:
No need to create the whole rev, what about: `next(filter(lambda x: x[1] == self, source._fields.items()), None)`?

Reply:
I think I'm just going to store fields_rev in matrixtable/table instead!

Reply:
OK, great!  I'd call it fields_inverse rather than reverse (reverse sounds like the keys are in the reverse order, this is inverting it in the sense of inverting a dict-as-function).

Reply:
Or `field_expr_name`?

Wait, why even look up the field, why not just check if self._ast is a Reference?

Reply:
this feels easier.

Reply:
I also don't want too much to depend on AST so we can change it.